sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -4158,7 +4158,8 @@ case class ArrayIntersect(left: Expression, right: Expression) extends ArrayBina
              |$nullElementIndex = $size;
              |$size++;
              |$builder.$$plus$$eq($nullValueHolder);
-           """.stripMargin)
+           """.stripMargin,
+          Some(s"$hashSet.containsNull()"))
```

Comment:
The interpreted code path is:
```
          val withArray1NullCheckFunc = SQLOpenHashSet.withNullCheckFunc(elementType, hsResult,
            (value: Any) => withArray1NaNCheckFunc(value),
            () =>
              if (hs.containsNull()) {
                arrayBuffer += null
              }
          )
```
Can we follow it in the codegen path? e.g. `handleNull` should check `if (hashSet.containsNull())`

Reply:
Updated